src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -513,11 +537,192 @@ internal void ReadToken(string encodedJson)
             EncodedToken = encodedJson;
         }
 
+#if NET6_0_OR_GREATER
+        /// <summary>
+        /// Converts a span into an instance of <see cref="JsonWebToken"/>.
+        /// </summary>
+        /// <param name="encodedJsonSpan">A span representing a 'JSON Web Token' (JWT) in JWS or JWE Compact Serialization Format.</param>
+        /// <exception cref="SecurityTokenMalformedException">if <paramref name="encodedJsonSpan"/> is malformed, a valid JWT should have either 2 dots (JWS) or 4 dots (JWE).</exception>
+        /// <exception cref="SecurityTokenMalformedException">if <paramref name="encodedJsonSpan"/> does not have an non-empty authentication tag after the 4th dot for a JWE.</exception>
+        /// <exception cref="SecurityTokenMalformedException">if <paramref name="encodedJsonSpan"/> has more than 4 dots.</exception>
+        internal void ReadToken(ReadOnlySpan<char> encodedJsonSpan)
+        {
+            // JWT must have 2 dots for JWS or 4 dots for JWE (a.b.c.d.e)
+            ReadOnlySpan<char> encodedJsonSpanCopy = encodedJsonSpan;
```

Comment:
Do we need these copies?

Reply:
The EncodedJsonSpanCopy is sliced multiple times and updated each time to extract different token parts, such as the header, payload, and authentication tag etc. The original EncodedJsonSpan is used at the end to store the underlying string into the EncodedToken property. At the moment, I cannot see a way around having a copy.

Reply:
Removed copies where possible.